velox/common/memory/MemoryPool.h

```diff
@@ -321,14 +321,8 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
   virtual int64_t capacity() const = 0;
 
   /// Returns the currently used memory in bytes of this memory pool. For
-  /// non-leaf memory pool, the function returns the aggregated memory
-  /// reservation from all its child memory pools.
-  virtual int64_t currentBytes() const = 0;
-
-  /// Returns the currently used memory in bytes of this memory pool. It returns
-  /// the same usage as currentBytes() for leaf memory pool. But for non-leaf
-  /// memory pool, the function returns the aggregated used memory from all its
-  /// child memory pools.
+  /// non-leaf memory pool, the function returns the aggregated used memory from
+  /// all its child memory pools.
   virtual int64_t usedBytes() const = 0;
```

Comment:
Curious why we don't need lock protection for used bytes?

Reply:
It might cause deadlock in some cases such as statsLocked(). We don't need lock for such individual counter access.